test/integration/src/test/java/io/pravega/test/integration/ReaderGroupNotificationTest.java

```diff
@@ -189,19 +213,128 @@ public void testSegmentNotifications() throws Exception {
+                    throw new IllegalStateException(e);
+                });
+        try {
+            Thread.sleep(2000);
+            scale(controller, SCOPE, streamName, 1, 1, threadPoolExecutor);
+            Thread.sleep(2000);
+        } catch (InterruptedException e) {
+            log.error("Interrupted: ", e);
+        }
+        Retry.withExpBackoff(10, 10, 100, 10000)
+                .retryingOn(IllegalStateException.class)
+                .throwingOn(RuntimeException.class)
+                .runAsync(() -> controller.getCurrentSegments(SCOPE, streamName)
+                        .thenAccept(streamSegments -> {
+                            try {
+                                Thread.sleep(1000);
+                                EventStreamReader<String> reader1 = clientFactory.createReader("readerId", "reader", new JavaSerializer<>(),
+                                        ReaderConfig.builder().initialAllocationDelay(0).build());
+                                for (int i = 0; i < NUM_EVENTS; i++) {
+                                    EventRead<String> result =  reader1.readNextEvent(1000);
+                                    assertNotNull(result);
+                                    assertTrue(result.getEvent().equals("data"));
+                                }
+                                val notificationResults = new ArrayBlockingQueue<SegmentNotification>(10);
+                                //Add segment event listener
+                                Listener<SegmentNotification> l1 = notification -> {
+                                    log.info("Number of Segments: {}, Number of Readers: {}", notification.getNumOfSegments(), notification.getNumOfReaders());
+                                    notificationResults.add(notification);
+                                };
+                                SegmentNotifier segmentNotifier = (SegmentNotifier) readerGroup.getSegmentNotifier(executor);
+                                segmentNotifier.registerListener(l1);
+                                segmentNotifier.pollNow();
+                                SegmentNotification initialSegmentNotification = notificationResults.poll();
+                                log.info("notification={}", initialSegmentNotification);
+                                readerGroup.resetReaderGroup(readerGroupConfig);
+                                readerGroup.readerOffline("readerId", null);
+                                long segmentSize = streamSegments.getSegments().size();
+                                log.info("segment_size={}", segmentSize);
+                                if (segmentSize > 3) {
+                                    log.info("success");
+                                    return;
+                                }
+                                throw new IllegalStateException("expected number of segments to be > 3");
+                            } catch (InterruptedException e) {
```

Comment:
This is an incorrect way of handling `InterruptedException`.

Use `Exceptions.handleInterrupted(...)` on whichever call declares that it's throwing this and remove this catch clause.

Reply:
Yes Andre